apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -334,101 +346,100 @@ export const OAuth2Form = () => {
+                    </li>
+                  )
+                })}
+              </ul>
+            )}
+            <div className="margin-top-20">
+              <Typography variant="body1">
+                By clicking <strong>Allow</strong>, you allow this app to use
+                your information in accordance with their{' '}
+                <Link
+                  href={oauthClientInfo.tos_uri}
+                  target="_blank"
+                  rel="noopener noreferrer"
                 >
-                  Deny
-                </Button>
-                <Button variant="contained" color="primary" onClick={onConsent}>
-                  Allow
-                </Button>
-              </div>
-            </StyledInnerContainer>
-          </StyledOuterContainer>
+                  terms of service
+                </Link>{' '}
+                and{' '}
+                <Link
+                  href={oauthClientInfo.policy_uri}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  privacy policy
+                </Link>
+                .
+              </Typography>
+            </div>
+            <div className="text-align-right margin-top-40">
+              <Button
+                variant="outlined"
+                onClick={onDeny}
+                sx={{ marginRight: '10px' }}
+              >
+                Deny
+              </Button>
+              <Button variant="contained" color="primary" onClick={onConsent}>
+                Allow
+              </Button>
+            </div>
+          </StyledInnerContainer>
         )}
       {isLoading && loadingSpinner}
-      {!redirectURL &&
-        !error &&
-        !accessToken &&
-        oauthClientInfo &&
-        oauthClientInfo.verified &&
-        oidcRequestDescription && (
-          <StyledOuterContainer>
-            <Paper sx={{ width: '400px', padding: '30px', margin: '0 auto' }}>
-              <StandaloneLoginForm
-                sessionCallback={() => {
-                  getSession()
-                }}
-                onBeginOAuthSignIn={() => {
-                  // save current route (so that we can go back here after SSO)
-                  localStorage.setItem(
-                    'after-sso-login-url',
-                    window.location.href,
-                  )
-                }}
-              />
-            </Paper>
-          </StyledOuterContainer>
-        )}
-      {error && (
-        <StyledOuterContainer>
-          <FullWidthAlert
-            variant="danger"
-            title={error.name || 'Error'}
-            description={`${error.reason} : ${error.message}`}
-            isGlobal={false}
+      {(!!twoFactorAuthErrorResponse ||
+        (!redirectURL &&
+          !error &&
+          !accessToken &&
+          oauthClientInfo &&
+          oauthClientInfo.verified &&
+          oidcRequestDescription)) && (
```

Comment:
In retrospect, I think we'd have a much easier time if we just had a separate route for authentication, though we'd need to add that route to our list of valid Google SSO redirect URLs

Reply:
It's a trivial task to add Google SSI redirect URLs, so let me know if you'd like to go forward with this idea